backend/src/test/java/com/google/minicrm/servlets/FormsServletTest.java

```diff
@@ -0,0 +1,283 @@
+import java.io.StringWriter;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+/**
+ * Provides Unit Tests for the FormsServlet servlet at endpoint api/forms.
+ */
+@RunWith(JUnit4.class)
+public final class FormsServletTest {
+
+  private static final String TEST_USER_ID = "testUserId";
+  private static final FormsServlet formsServlet = new FormsServlet();
+  private static Map<String, Object> envAttributes;
+  static {
+    envAttributes = new HashMap<>();
+    envAttributes.put("com.google.appengine.api.users.UserService.user_id_key", TEST_USER_ID);
+  }
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig(),
+          new LocalUserServiceTestConfig())
+          .setEnvIsLoggedIn(true)
+          .setEnvEmail("test@test.com")
+          .setEnvAuthDomain("testAuthDomain")
+          .setEnvAttributes(envAttributes);
+  private final Gson gson = new Gson();
+
+  private Form form1;
+  private Form form2;
+  private Form form3;
+  private HttpServletRequest request;
+  private HttpServletResponse response;
+
+  @Before
+  public void setUp() throws Exception {
+    helper.setUp();
+    request = mock(HttpServletRequest.class);
+    response = mock(HttpServletResponse.class);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void formsServlet_getRequest_returnsAllFormsInTimeOrder() throws Exception {
+    slowSeedForms();
+    Form[] returnedForms = getForms();
+    Form[] expectedForms = {form3, form2, form1};
+    assertArrayEquals(expectedForms, returnedForms);
+  }
+
+  @Test
+  public void formsServlet_putRequest_json_successfullyRenamesAndReturns204() throws Exception {
+    seedForms();
+    when(request.getContentType()).thenReturn("application/json;");
+    Reader reader = new StringReader(new FormsPutRequest("2", "newName").toJson());
+    when(request.getReader()).thenReturn(new BufferedReader(reader));
+    formsServlet.doPut(request, response);
+
+    //get the new forms
+    List<Form> returnedForms = Arrays.asList(getForms());
+    form2.setFormName("newName");
+    List<Form> expectedForms = new ArrayList<>(3);
+    expectedForms.add(form1);
+    expectedForms.add(form2);
+    expectedForms.add(form3);
+    //assert all forms are the same regardless of order in list
+    assertTrue(expectedForms.size() == returnedForms.size());
+    assertTrue(expectedForms.containsAll(returnedForms) && expectedForms.containsAll(returnedForms));
```

Comment:
This is to check lists containing equivalent form objects without regarding order. Is there a better way to do this? I tried to do assertEquals with two sets but it didn't seem to work (not sure why)